testsuite/forge/src/interface/public.rs

```diff
@@ -44,19 +45,41 @@ impl<'t> PublicUsageContext<'t> {
         self.public_info.chain_id
     }
 
+    pub fn tx_factory(&self) -> TransactionFactory {
+        self.public_info.tx_factory.clone()
+    }
```

Comment:
instead of cloning return a `&TransactionFactory` and let the caller determine if they want to clone it